### PR TITLE
Sites Dashboard v2: Update the table styles

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/6708, https://github.com/Automattic/dotcom-forge/issues/6709

## Proposed Changes

* Create the `client/components/dataviews` to share the styles of the dataviews component

image

image

## Testing Instructions

**Calypso**

* Go to /sites?flags=layout%2Fdotcom-nav-redesign-v2
* Make sure the table styles look good on both desktop and mobile

**A4A

* Go to /sites
* Make sure it works as before

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/components/items-dashboard/items-dataviews/a4a-style.scss
Comment (arthur791004) [inline client/a8c-for-agencies/components/items-dashboard/items-dataviews/a4a-style.scss:4]: Moved to client/components/dataviews/style.scss

File: client/a8c-for-agencies/components/items-dashboard/items-dataviews/style.scss
Comment (arthur791004) [inline client/a8c-for-agencies/components/items-dashboard/items-dataviews/style.scss:4]: Moved to client/components/dataviews/style.scss

File: client/a8c-for-agencies/components/items-dashboard/items-dataviews/style.scss
Comment (arthur791004) [inline client/a8c-for-agencies/components/items-dashboard/items-dataviews/style.scss:74]: Moved to client/components/dataviews/style.scss

File: client/a8c-for-agencies/components/layout/style.scss
Comment (arthur791004) [inline client/a8c-for-agencies/components/layout/style.scss:136]: Use `row-gap` to prevent it from misaligning the action buttons

**Review by cleacos — CHANGES_REQUESTED** [review]
At first glance, it looks like a good improvement, I like it.

Thank you for going ahead and doing it.

Since it's a core/foundation change, I have to review it well. So I have to block this PR until I finish some update/merges.

File: client/a8c-for-agencies/components/items-dashboard/items-dataviews/a4a-style.scss
Comment (cleacos) [reply client/a8c-for-agencies/components/items-dashboard/items-dataviews/a4a-style.scss:4]: I'm going to move it to another place. This should be in the A4A context.

File: client/a8c-for-agencies/components/items-dashboard/items-dataviews/style.scss
Comment (cleacos) [inline client/a8c-for-agencies/components/items-dashboard/items-dataviews/style.scss]: Is there any reason to leave these classes here?

**Review by cleacos — APPROVED** [review]
I pushed some commits. Moving classes between files and a small fix.

LGTM!

**Comment by arthur791004** [comment]
Nice, thank you!

**Comment by arthur791004** [comment]
> Remove vertical alignment affecting to the A4A column headers https://github.com/Automattic/wp-calypso/pull/89890/commits/9b4e15ffbf3cba183cffeab50294350053ec71cf

@cleacos I have a question about this. I think the star icon is not aligned vertically on the header without this change, and it also increases the height of the header. Is there any issue I didn't notice? Thank you!

| | Screenshots |
| - | - |
| Without vertical alignment  | Screenshot 2024-04-26 at 12 25 40 PM |
| With vertical alignment  | Screenshot 2024-04-26 at 12 24 38 PM |

**Comment by cleacos** [comment]
for some reason, at A4A, the SITE sorting icon was unaligned with the vertical alignment...

Thank you for the head up

**Comment by arthur791004** [comment]
I see. Let me think how to resolve it 🤔

**Comment by arthur791004** [comment]
Looks like adding `vertical-align: inherit` to svg element and removing the `vertical-align: top` from the fav icon will resolve the issue

**Comment by cleacos** [comment]
Fixed here: https://github.com/Automattic/wp-calypso/pull/89954

File: client/a8c-for-agencies/sections/sites/sites-dashboard/sites-dataviews-style.scss
Comment (fushar) [inline client/a8c-for-agencies/sections/sites/sites-dashboard/sites-dataviews-style.scss:28]: Somehow, sometimes this font size is used (12px) and sometimes this one is used (14px), which is the correct one?

https://github.com/Automattic/wp-calypso/blob/d1021f7b28ca9ece95f52015b80548942ea5165e/client/sites-dashboard/components/sites-site-url.ts#L10

Comment (cleacos) [reply client/a8c-for-agencies/sections/sites/sites-dashboard/sites-dataviews-style.scss:28]: I guess for Dotcom, it will be 14px. At A4A, we don't have the `SiteUrl` component.

**Pull request merged**
